### PR TITLE
[RAILS4] fixes #7594 - extract params defined in apipie documentation to use for strong parameters

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
@iNecas, can you take a look at `app/services/apipie_parser.rb` The idea is this:

```
> ApipieParser.allow_params('operatingsystems', 'update')
 => [:name, :major, :minor, :description, :family, :release_name] 

ApipieParser.allow_params('hostgroups', 'create')
 => [:name, :parent_id, :environment_id, :operatingsystem_id, :architecture_id, :medium_id, :ptable_id, :puppet_ca_proxy_id, :subnet_id, :domain_id, :realm_id, :puppet_proxy_id] 
```

File: app/services/apipie_parser.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  def self.method_descriptions(controller, version = Apipie.configuration.default_version)
+    Apipie.app.resource_descriptions[version][controller].method_descriptions
+  end
+
+  def self.allowed_params(controller, action, version = Apipie.configuration.default_version)
+    attributes = []
+    a = self.method_descriptions(controller).detect { |md| md.method.to_s == action.to_s }
+    a.params_ordered.each do |b|
+      if b.validator.try(:params_ordered)
```
Comment (isratrade) [inline app/services/apipie_parser.rb]: @iNecas, it took a while to figure out that the `validator` on a parameter is how to extract the sub parameters of a Hash parameter. Is there a different way to extract the POST/PUT params documented in the apipie dsl?

Comment (iNecas) [inline app/services/apipie_parser.rb]: I would probably go for Apipie.to_json(params[:version], params[:resource], params[:method]), in which case the params should be there inside a hash structure.

**Comment by isratrade** [comment]
@iNecas, I added  new commit using `Apipie.to_json` for your review.

File: app/services/apipie_parser.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Apipie.reload_documentation if Apipie.configuration.reload_controllers?
+
+class ApipieParser
+
+  delegate :logger, :to => :Rails
+
+  def self.allowed_params(controller, action, version = Apipie.configuration.default_version)
+     attributes = []
+     a = Apipie.to_json(version, controller, action)
+     a[:docs][:resources][0][:methods][0][:params].each do |b|
```
Comment (iNecas) [inline app/services/apipie_parser.rb:10]: You should be checing on the a not being nil, as that happened, if the particular action is not documented

**Comment by isratrade** [comment]
closing for now. probably want implement `strong_parametrs` in API v2 as it could potentially break what has been declared stable if we don't specify all possible attributes.

**Pull request closed without merging**